shared/network/2.x.md

```diff
@@ -362,6 +362,55 @@ settings.AddConnection(con)
 
 The above example will add a new WiFi connection file to your host OS **NetworkManager** configuration in `/etc/NetworkManager/system-connections`. It should be noted that you will not see a new file created in `resin-boot/system-connections` because there isn't a two-way binding. On device boot, the files in `resin-boot/system-connections` are copied into `/mnt/state/root-overlay/etc/NetworkManager/system-connections` without removing any files from the destination. That directory is then bind mounted to `/etc/NetworkManager/system-connections` in the root filesystem on the host.
 
+```
+#!/usr/bin/env python
+
+import dbus
+import ipaddress
+
+bus = dbus.SystemBus()
+
+# Get base NetworkManager dbus proxy
+nm_proxy = bus.get_object("org.freedesktop.NetworkManager", "/org/freedesktop/NetworkManager")
+nm_props = dbus.Interface(nm_proxy, "org.freedesktop.DBus.Properties")
+
+# Enumerate all devices
+devices = nm_props.Get("org.freedesktop.NetworkManager", "Devices")
+for device in devices:
+    # Get dbus proxy for the device
+    device_proxy = bus.get_object("org.freedesktop.NetworkManager", device)
+    device_props = dbus.Interface(device_proxy, "org.freedesktop.DBus.Properties")
+
+    # Get IPv4 configuration details
+    ipv4_path = device_props.Get("org.freedesktop.NetworkManager.Device", "Ip4Config")
+
+    # Skip devices that do not have IPv4 configured
+    if not ipv4_path or ipv4_path == "/":
+        continue
+
+    # Get dbus proxy for IPv4 configuration
+    ipv4_proxy = bus.get_object("org.freedesktop.NetworkManager", ipv4_path)
+    ipv4_props = dbus.Interface(ipv4_proxy, "org.freedesktop.DBus.Properties")
+
+    # List all IPv4 addresses
+    ipv4_addresses = ipv4_props.Get("org.freedesktop.NetworkManager.IP4Config", "Addresses")
+    addresses = []
+    for address in ipv4_addresses:
+        # address is an array where address[0] is the IPv4 address and address[1] is the netmask
+        # The IPv4 addresses are returned as integers, we need to flip endianity
+        addr_bytes = address[0].to_bytes(4, "little")
+        addr = ipaddress.IPv4Address(addr_bytes)
+        addresses.append("%s/%s" % (addr.compressed, address[1]))
+
+    # Get device name
+    device_name = device_props.Get("org.freedesktop.NetworkManager.Device", "Interface")
+
+    # Print all the information for a single device
+    print("%s: %s" % (device_name, ", ".join(addresses)))
+```
+
+The above example will print device name and all associated IPv4 addresses for each network device with an IPv4 connection.
```

Comment:
The above line might be better off as the one being before the code snippet to clearly explain what the code snippet is all about. This way folks know what they are looking at before scrolling further down and looking for explanations.

Reply:
I would say the same for the example above it, as having a description before the example provides much needed context to the code in the example.

Reply:
Moved both descriptions above